### PR TITLE
Midline ussd flows

### PR DESCRIPTION
**Pull request by fritzbrand**

### CODE REVIEW

File: mqr/midline_ussd.py [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+from mqr import config
+from vaccine.base_application import BaseApplication
+from vaccine.states import Choice, ChoiceState, EndState
+from vaccine.utils import HTTP_EXCEPTIONS, normalise_phonenumber
+
+logger = logging.getLogger(__name__)
+# TODO: FWB
+
+
+def get_eventstore():
```
Comment (erikh360) [inline mqr/midline_ussd.py:15]: We can remove this.

File: mqr/midline_ussd.py [hunk 2]
```diff
@@ -0,0 +1,689 @@
...
+        headers={
+            "Authorization": f"Token {config.RAPIDPRO_TOKEN}",
+            "Content-Type": "application/json",
+            "User-Agent": "mqr-midline-study-ussd",
+        },
+    )
+
+
+class Application(BaseApplication):
+    START_SURVEY = "state_start"
```
Comment (erikh360) [inline mqr/midline_ussd.py:43]: This should stay START_STATE and you can change it to state_eat fruits

File: mqr/midline_ussd.py [hunk 3]
```diff
@@ -0,0 +1,689 @@
...
+
+    async def state_start(self):
```
Comment (erikh360) [inline mqr/midline_ussd.py:45]: You can remove state_start and state_check_existing_result

File: mqr/midline_ussd.py [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+        return ChoiceState(
+            self,
+            question=question,
+            error=error,
+            choices=choices,
+            next="state_end",
+        )
+
+    # TODO FWB - Update all functions below this line (copied from baseline)
+    async def state_submit_data(self):
```
Comment (erikh360) [inline mqr/midline_ussd.py:571]: Can be removed, the last question should go to state_update_rapidpro_contact

File: mqr/midline_ussd.py [hunk 5]
```diff
@@ -0,0 +1,689 @@
...
+
+    async def state_update_rapidpro_contact(self):
+        msisdn = normalise_phonenumber(self.inbound.from_addr)
+        urn = f"whatsapp:{msisdn.lstrip(' + ')}"
+
+        async with get_rapidpro() as session:
+            for i in range(3):
+                try:
+                    data = {
+                        "flow": config.RAPIDPRO_BASELINE_SURVEY_COMPLETE_FLOW,
```
Comment (erikh360) [inline mqr/midline_ussd.py:621]: Rename to midline

**Review by erikh360 — COMMENTED** [review]
Some comments but looking good so far

File: mqr/midline_ussd.py (hunk 5 above)
Comment (fritzbrand) [reply mqr/midline_ussd.py:621]: Done

File: mqr/midline_ussd.py (hunk 4 above)
Comment (fritzbrand) [reply mqr/midline_ussd.py:571]: Done.  Not sure if you meant just the "state_submit_data" function, but I also removed "state_already_completed" and "state_contact_not_found" as they weren't being called anywhere I could see

File: mqr/midline_ussd.py (hunk 3 above)
Comment (fritzbrand) [reply mqr/midline_ussd.py:45]: Done

File: mqr/midline_ussd.py (hunk 2 above)
Comment (fritzbrand) [reply mqr/midline_ussd.py:43]: Not 100% sure if I understood you correctly, please double check if I got it right

File: mqr/midline_ussd.py (hunk 1 above)
Comment (fritzbrand) [reply mqr/midline_ussd.py:15]: Done

**Pull request merged**